### PR TITLE
registry: implement garbage collection subcommands

### PR DESCRIPTION
**Pull request by waynr**
In support of the new garbage collection feature for DOCR I am introducing the following new subcommands:

* `doctl registry garbage-collection start [<registry-name>]`
  * initiate garbage collection on a registry
  * will fail if a garbage collection has already been started
* `doctl registry garbage-collection get-active [<registry-name>]`
  * returns information about the currently-active garbage collection for a registry
* `doctl registry garbage-collection list [<registry-name>]`
  * lists all garbage collections that have been requested for a registry
* `doctl registry garbage-collection cancel [<registry-name>] <gc-uuid>`
  * cancels the specified garbage collection
  * requires an explicit gc to be specified to avoid accidentally cancelling the wrong garbage collection; unlikely currently, but may become an issue in the future when we support multiple registries.

I am opening this PR before finishing the tests so as to provide an opportunity as early as possible for reviewers to provide feedback while I work on the tests.

TODO:

* unit tests
  * [x] RunStartGarbageCollection
  * [x] RunGetGarbageCollection
  * [x] RunListGarbageCollection
  * [x] RunUpdateGarbageCollection
* integration tests
  * [x] `doctl registry garbage-collection start`
  * [x] `doctl registry garbage-collection get-active`
  * [x] `doctl registry garbage-collection list`
  * [x] `doctl registry garbage-collection cancel`
* [ ] deploy new API endpoints on the backend after it is clear that no more changes will be necessary in response to doctl/godo feedback
* [x] merge [associated godo PR](https://github.com/digitalocean/godo/pull/400)

### CODE REVIEW

File: commands/registry.go [hunk 1]
```diff
@@ -169,6 +170,77 @@ func Repository() *Command {
 	return cmd
 }
 
+// GarbageCollection creates the garbage-collection subcommand
+func GarbageCollection() *Command {
+	cmd := &Command{
+		Command: &cobra.Command{
+			Use:     "garbage-collection",
+			Aliases: []string{"gc", "g"},
+			Short:   "Display commands for working with garbage collection for a container registry",
```
Comment (sharmita3) [inline commands/registry.go:179]:
```suggestion
			Short:   "Display commands for garbage collection for a container registry",
```

File: commands/registry.go [hunk 2]
```diff
@@ -169,6 +170,77 @@ func Repository() *Command {
...
+			Long:    "The subcommands of `doctl registry garbage-collection` help you command actions related to garbage collecting unreferenced blobs for a specified registry.",
```
Comment (sharmita3) [inline commands/registry.go:180]:
```suggestion
			Long:    "The subcommands of `doctl registry garbage-collection` start a garbage collection, retrieve or cancel a currently-active garbage collection or list past garbage collections for a specified registry.",
```

File: commands/registry.go [hunk 3]
```diff
@@ -169,6 +170,77 @@ func Repository() *Command {
...
+  - UUID
+  - Status
+  - Registry Name
+  - Created At
+  - Updated At
+  - Blobs Deleted
+  - Freed Bytes
+`
+
+	runGetGarbageCollectionDesc := "This command retrieves a container registry's currently-active garbage collection, if any active garbage collection exists. Information included about the registry includes:" + gcInfoIncluded
```
Comment (sharmita3) [inline commands/registry.go:206]:
```suggestion
	runGetGarbageCollectionDesc := "This command retrieves the currently-active garbage collection for a container registry, if any active garbage collection exists. Information included about the registry includes:" + gcInfoIncluded
```

File: commands/registry.go [hunk 4]
```diff
@@ -169,6 +170,77 @@ func Repository() *Command {
...
+	CmdBuilder(
+		cmd,
+		RunGetGarbageCollection,
+		"get-active",
+		"Retrieve information about the currently-active garbage collection.",
```
Comment (sharmita3) [inline commands/registry.go:211]:
```suggestion
		"Retrieve information about the currently-active garbage collection for a container registry",
```

File: commands/registry.go [hunk 5]
```diff
@@ -169,6 +170,77 @@ func Repository() *Command {
...
+		runGetGarbageCollectionDesc,
+		Writer,
+		aliasOpt("ga", "g"),
+		displayerType(&displayers.GarbageCollection{}),
+	)
+
+	runListGarbageCollectionsDesc := "This command retrieves a list of past garbage collections for a given garbage collection. Information about each garbage collection includes:" + gcInfoIncluded
```
Comment (sharmita3) [inline commands/registry.go:218]:
```suggestion
	runListGarbageCollectionsDesc := "This command retrieves a list of past garbage collections for a registry. Information about each garbage collection includes:" + gcInfoIncluded
```

File: commands/registry.go [hunk 6]
```diff
@@ -169,6 +170,77 @@ func Repository() *Command {
...
+	CmdBuilder(
+		cmd,
+		RunListGarbageCollections,
+		"list",
+		"Retrieve information about list of past collections",
```
Comment (sharmita3) [inline commands/registry.go:223]:
```suggestion
		"Retrieve information about past garbage collections for a container registry",
```

File: commands/registry.go [hunk 7]
```diff
@@ -169,6 +170,77 @@ func Repository() *Command {
...
+		runListGarbageCollectionsDesc,
+		Writer,
+		aliasOpt("ls", "l"),
+		displayerType(&displayers.GarbageCollection{}),
+	)
+
+	runCancelGarbageCollectionDesc := "This command cancels a registry's currently-active garbage collection."
```
Comment (sharmita3) [inline commands/registry.go:230]:
```suggestion
	runCancelGarbageCollectionDesc := "This command cancels the currently-active garbage collection for a container registry."
```

File: commands/registry.go [hunk 8]
```diff
@@ -169,6 +170,77 @@ func Repository() *Command {
...
+	CmdBuilder(
+		cmd,
+		RunCancelGarbageCollection,
+		"cancel",
+		"Cancel the currently-active garbage collection",
```
Comment (sharmita3) [inline commands/registry.go:235]:
```suggestion
		"Cancel the currently-active garbage collection for a container registry",
```

**Review by adamwg — APPROVED** [review]
lgtm.

File: errors.go [hunk 9]
```diff
@@ -30,3 +30,19 @@ func NewMissingArgsErr(cmd string) *MissingArgsErr {
 func (e *MissingArgsErr) Error() string {
 	return fmt.Sprintf("(%s) command is missing required arguments", e.Command)
 }
+
+// TooManyArgsErr is returned when there are too many arguments for a command.
```
Comment (andrewsomething) [inline errors.go:34]: Thanks for adding this! There are quite a few places where we should be taking advantage of a common error for this. 

```
$ grep -r "if len(c.Args) != 1" commands/ | wc -l
76
```

**Review by andrewsomething — APPROVED** [review]
:+1: LGTM

**Review by andrewsomething — APPROVED** [review]

**Pull request merged**